release-calendar/src/client/components/ChannelTable.tsx

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '@material/react-list/index.scss';
+import * as React from 'react';
+import {DATA} from '../models/data';
+import {getMostRecentChannelRTVs} from '../models/release-channel';
+import List, {
+  ListDivider,
+  ListGroup,
+  ListGroupSubheader,
+  ListItem,
+  ListItemText,
+} from '@material/react-list';
+
+export interface ChannelTableProps {
+  // TODO: considering moving the RTVs out of the ChannelTable Component altogether and
+  // intead having them live directly next door. Still testing.
+  handleSelectChannel: (selected: number) => void;
+  selectedChannel: number;
+}
+
+export interface ChannleTableState {
+  rtvs: Map<string, string>;
+}
+
+// eslint-disable-next-line prettier/prettier
+export class ChannelTable extends React.Component<ChannelTableProps, ChannleTableState> {
+  state: ChannleTableState;
+  constructor(props: Readonly<ChannelTableProps>) {
+    super(props);
+    this.state = {
+      rtvs: new Map(),
+    };
+    this.handleChannelChange = this.handleChannelChange.bind(this);
+  }
+
+  componentDidMount(): void {
+    Promise.resolve(getMostRecentChannelRTVs(DATA)).then(result =>
```

Comment:
You said in another comment that DATA includes all releases and the calendar does the filtering for you. That means that DATA doesn't necessary have only the current releases, which `getMostRecentChannelRTVs` depends on. I think we need a separate API call here and not use DATA.

Reply:
I committed again. I still kept the call the same in `ChannelTable` but changed `release-channel` to expected an unordered DATA. I figured for development purposes I shouldnt make a whole new data file if the differences in the calls in the end will most likely be size. So while the data from the server differs later on, `DATA` can still get it done for both

Reply:
So will DATA always contain current releases? Even if the user is selecting a month from 2 years ago?

Reply:
because we separated the api calls per each component. yes. The channelTable component wont remount unless something within its own state or the state of the props passed down to it are updated.